caffe2/serialize/read_adapter_interface.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+namespace at {
+
+// this is the interface for the (file/stream/memory) reader in
+// PyTorchStreamReader. with this interface, we can extend the support
+// besides standard istream
+class ReadAdapterInterface {
+ public:
+  virtual size_t size() = 0;
+  virtual size_t read(uint64_t pos, void* buf, size_t n, const char* what) = 0;
```

Comment:
default `what` to nullptr?

Reply:
I will just set the default value to "" (I feel it's safer)